src/main/java/uk/gov/di/ipv/cri/common/library/service/SessionService.java

```diff
@@ -126,25 +129,42 @@ public SessionItem getSession(String sessionId) {
 
     public SessionItem getSessionByAccessToken(AccessToken accessToken)
             throws SessionExpiredException, AccessTokenExpiredException, SessionNotFoundException {
-        SessionItem sessionItem;
 
-        try {
-            sessionItem =
-                    ListUtil.getOneItemOrThrowError(
-                            dataStore.getItemByIndex(
-                                    SessionItem.ACCESS_TOKEN_INDEX,
-                                    accessToken.toAuthorizationHeader()));
-        } catch (IllegalArgumentException e) {
-            if (e.getMessage().contains("No items found")) {
-                throw new SessionNotFoundException("no session found with that access token");
-            } else {
-                throw new SessionNotFoundException(
-                        "more than one session found with that access token");
-            }
-        }
+        RetryConfig retryConfig =
+                new RetryConfig.Builder()
+                        .delayBetweenAttempts(100)
+                        .maxAttempts(3)
+                        .exponentiallyRetry(true)
+                        .build();
+
+        Retryable<SessionItem> retryable =
+                () -> {
+                    SessionItem sessionItem;
+
+                    try {
+                        sessionItem =
+                                ListUtil.getOneItemOrThrowError(
+                                        dataStore.getItemByIndex(
+                                                SessionItem.ACCESS_TOKEN_INDEX,
+                                                accessToken.toAuthorizationHeader()));
+                    } catch (IllegalArgumentException e) {
+                        if (e.getMessage().contains("No items found")) {
+                            throw new SessionNotFoundException(
+                                    "no session found with that access token");
```

Comment:
What do you think about putting some logger.errors in here?

Reply:
It'll bubble down: `LOGGER.error("Retry attempt {} failed", attempt, e);` to this line